### PR TITLE
On mac a difference error is thrown on connection refusal

### PR DESCRIPTION
**Pull request by ddustin**
@cdecker

### CODE REVIEW

File: tests/test_connection.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def test_connect(node_factory):
         l1.rpc.connect('032cf15d1ad9c4a08d26eab1918f732d8ef8fdc6abb9640bf3db174372c491304e')
 
     # Should get reasonable error if connection refuse.
-    with pytest.raises(RpcError, match=r'Connection establishment: Connection refused'):
+    with pytest.raises(RpcError, match=r'Connection establishment: (Connection refused|Bad file descriptor)'):
```
Comment (vincenzopalazzo) [inline tests/test_connection.py:56]: In this case, I usualy preferer to add a comment that points out that we can receive different err message for different os. However, maybe we could be interested only in the prefix of the error?

**Comment by rustyrussell** [comment]
This hints at an underlying bug, TBH.  ECONNREFUSED is definitely the error we expect...

**Comment by cdecker** [comment]
Testing on a Mac myself, it seems that something is up with these kinds of errors. Every once in a while I seem to be losing the `hsmd` filedescriptor, which is weird.

Still working on making compilation on m1 solid, but will keep an eye open on why this fails. In the meantime if there's any pointers as to how to debug this, they'd be much appreciated.

**Comment by rustyrussell** [comment]
> Testing on a Mac myself, it seems that something is up with these kinds of errors. Every once in a while I seem to be losing the hsmd filedescriptor, which is weird.

Hmm.  This is usually caused by someone closing the wrong fd (less often, by dup2 on top of the fd).  Try replacing close with closex which does `assert(!getenv("HSM_FD") || fd != atoi(getenv("HSM_FD"))` then setenv that to "3" at the start of connectd?

That will at least tell us who is closing it, if not why they're wrong.  If you can make it really reproducible you can debug the subd and poke around, but that's harder if it's intermittent.

**Comment by whitslack** [comment]
FWIW, I saw this on Linux and had the same gut reaction as @rustyrussell. "Bad file descriptor" indicates a logic bug: passing to a syscall an integer that does not specify an open file descriptor.

**Comment by cdecker** [comment]
I think we can close this, the root problem being somewhere else :-)

**Comment by ddustin** [comment]
Wfm 👍

**Pull request closed without merging**